src/pytorch_lightning/trainer/connectors/signal_connector.py

```diff
@@ -51,19 +51,28 @@ def register_signal_handlers(self) -> None:
         environment = self.trainer._accelerator_connector.cluster_environment
         if isinstance(environment, SLURMEnvironment) and environment.auto_requeue:
             log.info("SLURM auto-requeueing enabled. Setting signal handlers.")
-            sigusr1_handlers.append(self.slurm_sigusr1_handler_fn)
+            sigusr_handlers.append(self.slurm_sigusr_handler_fn)
             sigterm_handlers.append(self.sigterm_handler_fn)
 
-        # signal.SIGUSR1 doesn't seem available on windows
+        # Windows seems to have signal incompatibilities
         if not self._is_on_windows():
-            if sigusr1_handlers and not self._has_already_handler(signal.SIGUSR1):
-                self._register_signal(signal.SIGUSR1, HandlersCompose(sigusr1_handlers))
+            if isinstance(environment, SLURMEnvironment):
+                sigusr = getattr(signal, "SIG" + environment.signal, None)
+
+                if sigusr is None:
+                    log.warning(f"Unrecognized signal name SIG{environment.signal}, assuming SIGUSR1")
+                    sigusr = signal.SIGUSR1
```

Comment:
I suggest to move this validation to the SLURMEnvironment directly. What do you think?
Then, we can assume that SLURMEnvironment.signal will always return a good one.